CarSeatCoverSearch.java

```diff
@@ -0,0 +1,205 @@
+        String phoneNumberInput;
+        do {
+            phoneNumberInput = JOptionPane.showInputDialog(null,"Please enter your phone number. Format: 0412838475",appName, JOptionPane.QUESTION_MESSAGE);
+            if (phoneNumberInput == null) System.exit(0);
+            try {
+                phoneNumber = Long.parseLong(phoneNumberInput);
+            } catch (NumberFormatException n) {
+                JOptionPane.showMessageDialog(null, "Invalid entry. Please enter your 10 digit phone number.");
+            }
+        }while(phoneNumberInput.length()!=10);
+
+        return new Geek(name, phoneNumber);
+    }
+
+    /**
+     * generates JOptionPanes requesting user input for car seat cover type, theme, min and max price
+     * @return a CarSeatCover object representing the user's dream product
+     */
+    private static CarSeatCover getUserCriteria(){
+
+        // asking for users type
+        Type type = (Type) JOptionPane.showInputDialog(null,"Please select your preferred type:",appName, JOptionPane.QUESTION_MESSAGE,null,Type.values(),Type.U30);
+        if(type==null) System.exit(0);
+
+        //String userType = Type.valueOf(type);
+
+        // asking for users theme
+        String theme  = (String) JOptionPane.showInputDialog(null,"Please select your preferred theme.",appName, JOptionPane.QUESTION_MESSAGE,null, carSeatCovers.getAllThemes().toArray(), "");
+        if(theme==null) System.exit(0);
+
+        // asking for users min price
+        float minPrice = -1, maxPrice = -1;
+        while(minPrice==-1) {
+            try {
+                minPrice = Float.parseFloat(JOptionPane.showInputDialog(null,"Please enter the lowest price ",appName,JOptionPane.QUESTION_MESSAGE));
+            }
+            catch (NumberFormatException e){
+                JOptionPane.showMessageDialog(null,"Invalid input. Please try again.");
+            }
+        }
+
+        // asking for users max price
+        while(maxPrice<minPrice) {
+            try {
+                maxPrice = Float.parseFloat(JOptionPane.showInputDialog(null,"Please enter the highest price ",appName,JOptionPane.QUESTION_MESSAGE));
+            }
+            catch (NumberFormatException e){
+                JOptionPane.showMessageDialog(null,"Invalid input. Please try again.");
+            }
+            if(maxPrice<minPrice) JOptionPane.showMessageDialog(null,"Highest price must be >= lowest price.");
+        }
+
+        CarSeatCover dreamProduct = new CarSeatCover("", 0, -1, theme, Collections.singleton(type), "");
+        dreamProduct.setMinPrice(minPrice);
+        dreamProduct.setMaxPrice(maxPrice);
+        //dreamProduct.setUserType(userType);
+        return dreamProduct;
+    }
+
+    /**
+     * method to load all car seat cover data from file, using the data objects in an instance of Inventory
+     * @return an CarSeatCovers object - this is a dataset of the data objects
+     */
+    private static Inventory loadInventory() {
```

Comment:
Going back to that previous conversation on the `Inventory` class, you could consider the loading of this inventory as part of the `Inventory` class's job. But it could also not be.

Like maybe we don't consider it part of the inventory person's job to know how to process text and turn it into products, which is why we are doing it this way.